pkg/ingester/shipper_test.go

```diff
@@ -77,6 +82,11 @@ func TestShipper(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, 1, uploaded)
 
+		// Verify that the lastSuccessfulUploadTime was updated to within the last 2 seconds.
+		// WARNING(jhesketh): This check is ultimately non-deterministic. If we get flakiness
+		//										on this/in the CI etc, we should increase the duration.
```

Comment:
We're using this approach in a bunch of places. I would avoid having to add such comment everywhere. Also the 3nd looks bad formatted.

```suggestion
```